### PR TITLE
Google vision

### PR DESCRIPTION
**Pull request by pauloouriques**

### CODE REVIEW

File: components/OnboardingForm.js [hunk 1]
```diff
@@ -0,0 +1,365 @@
+import { useState } from "react";
+import Image from "next/image";
+import { Checkbox, Input, Button, Modal } from "@nextui-org/react";
+// import * as notify from "../../../bibliopal-nextjs/pages/api/notifier/notify";
```
Comment (pauloouriques) [inline components/OnboardingForm.js:4]: uncomment this

File: components/OnboardingForm.js [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+      if (!res.ok) {
+        throw new Error("Failed to onboard user");
+      }
+
+      const data = await res.json();
+      loadingHandler(false);
+      isCompleteHandler();
+    } catch (err) {
+      // notify.error(err)
```
Comment (pauloouriques) [inline components/OnboardingForm.js:223]: uncomment this

File: components/SignIn.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+import IconEnvelope from "../assets/svg/icons_envelope.svg";
+import Link from "next/link";
+
+const providers = [
+  {
+    name: "google",
+    icon: IconGoogle,
+  },
+  {
+    name: "facebook",
```
Comment (pauloouriques) [inline components/SignIn.js]: remove it, since it's not being used

File: components/SignIn.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    icon: "/icons/icon-fb.svg",
+  },
+  {
+    name: "apple",
```
Comment (pauloouriques) [inline components/SignIn.js]: remove it, since it's not being used

File: components/utility/Capture.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    loading = false
+}) {
+    const webcamRef = useRef(null);
+    const [facingMode, setFacingMode] = useState("environment");
+
+    useEffect(() => {
+        navigator.mediaDevices
+            .getUserMedia({ video: { facingMode: "environment" } })
+            .catch((error) => {
+
```
Comment (pauloouriques) [inline components/utility/Capture.js:18]: add error notify here

File: components/utility/ListingItem.js [hunk 6]
```diff
@@ -1,40 +1,128 @@
...
+        ))}
+        {isSlot && children ?
+          <div className='mt-1.5' onClick={stopClick}>
+            {children}
+          </div> : ''}
       </div>
+
+
     </div>
+    //   <div
```
Comment (pauloouriques) [inline components/utility/ListingItem.js]: remove unused code

File: components/utility/ProductDetails.js [hunk 7]
```diff
@@ -113,7 +113,7 @@ function ProductDetails({ open, onClose, data, fetchListings, imageOnly = false
                             )}
```
Comment (pauloouriques) [inline components/utility/ProductDetails.js]: wrong indentation/formatting. Please, add the prettier plugin to your IDE.

File: pages/api/add-listing.ts [hunk 8]
```diff
@@ -1,80 +1,75 @@
...
-            {
-              contentType: "image/png",
-            }
-          );
-        if (uploadResponse.error) {
-          throw new Error(uploadResponse.error.message);
-        }
-        const imagePath = uploadResponse.data.path;
+        case "admin":
+          payload = {
```
Comment (pauloouriques) [inline pages/api/add-listing.ts:47]: maybe:
`payload = Object.assign({}, listing, {type: 'admin'})`

File: pages/api/admin/fetchUsers.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import { prisma } from "../../../db/prismaDB";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
+
+const handler = async (req, res) => {
+  const session = await getServerSession(req, res, authOptions);
+
+  if (
+    session.user.email !== "neurogoop@gmail.com" &&
```
Comment (pauloouriques) [inline pages/api/admin/fetchUsers.js:9]: add this hardcoded emails to .env

File: pages/api/remove-image-background.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
+// import axios from 'axios';
```
Comment (pauloouriques) [inline pages/api/remove-image-background.js]: remove unused code

File: pages/api/upload-image.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
+import { createClient } from '@supabase/supabase-js';
```
Comment (pauloouriques) [inline pages/api/upload-image.js]: wrong formatting

File: pages/business/sales.js [hunk 12]
```diff
@@ -25,11 +25,10 @@ const ProfileComponent = ({ }) => {
```
Comment (pauloouriques) [inline pages/business/sales.js]: formatting

File: pages/consumer/saved.js [hunk 13]
```diff
@@ -239,7 +239,7 @@ export default function Home() {
                               ))}
                             </div>
                           </div>
```
Comment (pauloouriques) [inline pages/consumer/saved.js:240]: remove unused code

File: pages/login.js [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+            if (res?.ok) {
+                // Successful authentication
+                router.push("/business");
+
+            } else {
+                // Failed authentication
+                setError("Failed! Check your input and try again.");
+            }
+        } catch (error) {
+            setError("An error occurred. Please try again later.");
```
Comment (pauloouriques) [inline pages/login.js]: add notify

File: pages/register.js [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+
+        if (res.ok) {
+            const data = await res.json();
+
+            // registration success
+        } else {
+            //registration faled
+        }
+    }
+    return (
```
Comment (pauloouriques) [inline pages/register.js]: formatting

File: prisma/migrations/20231011113808_emplyee_and_admin_listing/migration.sql [hunk 16]
```diff
@@ -0,0 +1,16 @@
+-- AlterTable
```
Comment (pauloouriques) [inline prisma/migrations/20231011113808_emplyee_and_admin_listing/migration.sql]: remove all migrations

**Review by pauloouriques — COMMENTED** [review]
please, remove every model changes and migration from this PR and you also need to fix the conflicts

File: prisma/migrations/20231011113808_emplyee_and_admin_listing/migration.sql (hunk 16 above)
Comment (theibrahim) [reply prisma/migrations/20231011113808_emplyee_and_admin_listing/migration.sql]: done

File: pages/register.js (hunk 15 above)
Comment (theibrahim) [reply pages/register.js]: added plugin in vs code and run control+s in all files. Let me know if you want me change any settings in vs code configuration file?

File: pages/login.js (hunk 14 above)
Comment (theibrahim) [reply pages/login.js]: Added frontend notification

File: pages/consumer/saved.js (hunk 13 above)
Comment (theibrahim) [reply pages/consumer/saved.js:240]: done

File: pages/business/sales.js (hunk 12 above)
Comment (theibrahim) [reply pages/business/sales.js]: added plugin in vs code and run control+s in all files. Let me know if you want me change any settings in vs code configuration file?

File: pages/api/upload-image.js (hunk 11 above)
Comment (theibrahim) [reply pages/api/upload-image.js]: added plugin in vs code and run control+s in all files. Let me know if you want me change any settings in vs code configuration file?

File: pages/api/remove-image-background.js (hunk 10 above)
Comment (theibrahim) [reply pages/api/remove-image-background.js]: not doing backend

File: pages/api/admin/fetchUsers.js (hunk 9 above)
Comment (theibrahim) [reply pages/api/admin/fetchUsers.js:9]: its related to backend. you should do

File: pages/api/add-listing.ts (hunk 8 above)
Comment (theibrahim) [reply pages/api/add-listing.ts:47]: its related to backend. you should do

File: components/utility/ProductDetails.js (hunk 7 above)
Comment (theibrahim) [reply components/utility/ProductDetails.js]: done

File: components/utility/ListingItem.js (hunk 6 above)
Comment (theibrahim) [reply components/utility/ListingItem.js]: done

File: components/utility/Capture.js (hunk 5 above)
Comment (theibrahim) [reply components/utility/Capture.js:18]: done, frontend nofication

File: components/SignIn.js (hunk 4 above)
Comment (theibrahim) [reply components/SignIn.js]: done

File: components/SignIn.js (hunk 3 above)
Comment (theibrahim) [reply components/SignIn.js]: done

File: components/OnboardingForm.js (hunk 2 above)
Comment (theibrahim) [reply components/OnboardingForm.js:223]: for you because its related to backend

File: components/OnboardingForm.js (hunk 1 above)
Comment (theibrahim) [reply components/OnboardingForm.js:4]: for you

**Review by pauloouriques — COMMENTED** [review]
looks good, please merge it

**Pull request merged**